lab-katy/__test__/collection-router.test.js

```diff
@@ -0,0 +1,245 @@
+        .then( collection => {
+          this.tempCollection = collection;
+          done();
+        })
+        .catch(done);
+    });
+
+    afterEach( () => {
+      delete exampleCollection.userID;
+    });
+
+    it('should return a collection', done => {
+      request.get(`${url}/api/collection/${this.tempCollection._id}`)
+        .set({
+          Authorization: `Bearer ${this.tempToken}`,
+        })
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res.status).toEqual(200);
+          expect(res.body.name).toEqual(exampleCollection.name);
+          expect(res.body.desc).toEqual(exampleCollection.desc);
+          expect(res.body.userID).toEqual(this.tempUser._id.toString());
+          done();
+        });
+    });
+
+    it('should return a 401 if no token provided', done => {
+      request.get(`${url}/api/collection/${this.tempCollection._id}`)
+        .end((err, res) => {
+          expect(res.status).toEqual(401);
+          done();
+        });
+    });
+
+    it('should return a 404 if id not found', done => {
+      request.get(`${url}/api/collection/`)
+        .end((err, res) => {
+          expect(res.status).toEqual(404);
+          done();
+        });
+    });
+  });
+
+  //PUT ROUTE TESTS
+
+  describe('PUT: /api/collection/:collectionId', () => {
+    beforeEach( done => {
+      new User(exampleUser)
+        .generatePasswordHash(exampleUser.password)
+        .then( user => {
+          this.tempUser = user;
+          return user.generateToken();
+        })
+        .then( token => {
+          this.tempToken = token;
+          done();
+        })
+        .catch(done);
+    });
+
+    beforeEach( done => {
+      exampleCollection.userID = this.tempUser._id.toString();
+      new Collection(exampleCollection).save()
+        .then( collection => {
+          this.tempCollection = collection;
+          done();
+        })
+        .catch(done);
+    });
+
+    afterEach( () => {
+      delete exampleCollection.userID;
+    });
+
+    it('should return a collection', done => {
+      request.put(`${url}/api/collection/${this.tempCollection._id}`)
+        .send(updateCollection)
+        .set({
+          Authorization: `Bearer ${this.tempToken}`,
+        })
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res.status).toEqual(200);
+          expect(res.body.desc).toEqual(updateCollection.desc);
+          expect(res.body.name).toEqual(updateCollection.name);
+          expect(res.body.userID).toEqual(this.tempUser._id.toString());
+          done();
+        });
+    });
+
+    it('should return a 401 with no token', done => {
+      request.put(`${url}/api/collection/${this.tempCollection._id}`)
+        .send(updateCollection)
+        .end((err, res) => {
+          expect(res.status).toEqual(401);
+          done();
+        });
+    });
+
+    it('should return a 404 invalid id', done => {
```

Comment:
not an invalid id, this is just hitting express' built in catch all because it technically isn't hitting a route you've defined